packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -794,6 +799,97 @@ Static group authorization should perform as expected.`,
     );
   };
 
+  addAutoGeneratedFields = (
+    ctx: TransformerContextProvider,
+    def: ObjectTypeDefinitionNode,
+    allowedFields: string[],
+    fields: readonly string[],
+  ) => {
+    this.addAutoGeneratedRelationalFields(ctx, def, allowedFields, fields);
+    this.addAutoGeneratedIndexFields(def, allowedFields);
+  };
+
+  addAutoGeneratedRelationalFields = (
+    ctx: TransformerContextProvider,
+    def: ObjectTypeDefinitionNode,
+    allowedFields: string[],
+    fields: readonly string[],
+  ) => {
+    this.addAutoGeneratedHasManyFields(ctx, def, allowedFields);
+    this.addAutoGeneratedHasOneFields(ctx, fields, def, allowedFields);
+  };
+
+  addAutoGeneratedIndexFields = (def: ObjectTypeDefinitionNode, allowedFields: string[]) => {
+    const sortKeyFieldValues: ListValueNode[] = def.fields
+      ?.map(it => it.directives)
+      .flat()
+      .filter(it => it.name.value === 'primaryKey' || it.name.value === 'index')
+      .map(it => it.arguments)
+      .flat()
+      .filter(it => it.name.value === 'sortKeyFields' && it.value.kind === 'ListValue' && it.value.values.length > 1)
+      .map(it => it.value)
+      .flat() as ListValueNode[];
+
+    for (const sortKeyFieldValue of sortKeyFieldValues) {
+      const accessOnAllKeys = !sortKeyFieldValue.values.some(it => it.kind !== 'StringValue' || !allowedFields.includes(it.value));
+      if (!accessOnAllKeys) continue;
+
+      const keyName = sortKeyFieldValue.values.map(it => (it as StringValueNode).value).join(ModelResourceIDs.ModelCompositeKeySeparator());
+      if (!allowedFields.includes(keyName)) {
+        allowedFields.push(keyName);
+      }
```

Comment:
nit: might be more readable
```suggestion
      if(accessOnAllKeys) {
        const keyName = sortKeyFieldValue.values.map(it => (it as StringValueNode).value).join(ModelResourceIDs.ModelCompositeKeySeparator());
        if (!allowedFields.includes(keyName)) {
          allowedFields.push(keyName);
        }
      }
```

Reply:
updated